src/gennylib/src/dbcheck.cpp

```diff
@@ -0,0 +1,150 @@
+// Copyright 2021-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <atomic>
+#include <mutex>
+#include <thread>
+
+#include <bsoncxx/array/view.hpp>
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/builder/basic/kvp.hpp>
+#include <bsoncxx/types.hpp>
+#include <mongocxx/client.hpp>
+#include <mongocxx/database.hpp>
+#include <mongocxx/uri.hpp>
+
+#include <boost/log/trivial.hpp>
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/json.hpp>
+#include <gennylib/dbcheck.hpp>
+#include <gennylib/v1/Topology.hpp>
+
+// Logic in this file is based on the following two js implementations:
+// https://github.com/10gen/workloads/blob/aeaf42b86bb8f1af9bc6ac90198ac0b4ff32bd14/utils/mongoshell.js#L481
+// https://github.com/mongodb/mongo-perf/blob/bd8901a2e76d2fb13d2a6a313f7a9e1bf6be9c04/util/utils.js#L384-L387
+
+namespace genny {
```

Comment:
I don't think you need a separate header for this, the Actor is intended to be the interface for running the dbcheck logic. If we need to pull it out for other Actors to use, we can do so later when we have the two use-cases in mind (e.g. perhaps more precise coordination between `Orchestrator`/Phase lifecycle and the (nicely implemented) `WaitForDbCheckVisitor`.

Reply:
I was trying to follow the same design of the [QuiesceActor](https://github.com/mongodb/genny/blob/869fe58310e95d39b6e1d1a63b15f0723f8a2d0d/src/gennylib/src/quiesce.cpp) as we share the same design.

Reply:
Ya, that's a fair response. We indeed already have precedent for this in `quiesce`. I'm not sure why this pattern didn't set off a klaxon in my head when we added that.

I'm still rather partial to _not_ having library functions like this. It encourages very fine-grained levels of abstraction that quickly make Genny into a shadow API of the C++ driver.

There may be some notion of logic-re-use across Actors in the future, but I don't think we've found the right solution (or particular need) quite yet. Efforts to slim down / split / extend `CrudActor` might be a proving ground for that more generally.

Please consider moving this into the Actor's cpp. But I wouldn't blame you for wanting to move on and pushing back.

My recommendation--if you don't end up moving this to the Actor--would be to add a comment to both `quiesce` and `dbcheck` indicating that they are intended only for use by their respective Actors.

```c++
// Note: This library function is only intended to be used
//       by the DbCheck Actor.
//
//       If you need to run dbcheck in your workload, you can
//       include the DbCheckActor in your workload config.
//
//       If you believe you need to use this anywhere other
//       than DbCheckActor.{cpp,hpp} please work with the
//       Genny maintainers to discuss how to evolve this API.   
```

And similarly, please add at least high-level docs about what dbcheck (the function) does or ensure it directly references existing docs in the Actor part of the code.

Reply:
I'll also ping @PyGeek03 here. Ultimately he and the rest of his team are responsible for the future of Genny, so it's totally up to them how to handle this kind of scenario. I'm happy to give input / past experience, but there may be a good reason to encourage this kind of factorization in the future.

Reply:
I have already moved it to the Actor.